mypy.ini

```diff
@@ -1,44 +1,15 @@
 [mypy]
-python_version=3.7
-
-show_column_numbers=True
-show_error_codes=True
-
-# NOTE: ideally we'd have these all be set to True, but that requires a decent
-# number of changes
-disallow_any_unimported=False
-disallow_any_expr=False
-disallow_any_decorated=False
-disallow_any_explicit=False
-disallow_any_generics=True
-disallow_subclassing_any=True
-
-disallow_untyped_calls=False
-disallow_untyped_defs=True
-disallow_incomplete_defs=True
-check_untyped_defs=False
-disallow_untyped_decorators=True
-
-no_implicit_optional=True
-strict_optional=True
-
-warn_redundant_casts=True
-warn_unused_ignores=True
-warn_no_return=True
-warn_return_any=False
-warn_unreachable=False
-
-strict_equality=True
```

Comment:
I think we probably want to keep most of these due to the defaults for strictness options mostly being disabled:

https://mypy.readthedocs.io/en/stable/config_file.html#confval-strict_equality

I agree it's kind of icky, but one way to think of it is explicit over implicit

another thing, they may not error now, but they will help warn in the future

Reply:
Hmm, ok - understood that the checks we run could be made stricter.  What would the costs and benefits of stricter checks be to our developers and maintainers?